internal/pkg/symnotify/symnotify.go

```diff
@@ -0,0 +1,112 @@
+// package symnotify provides a file system watcher that notifies events for symlink targets.
+//
+package symnotify
+
+import (
+	"io"
+	"io/ioutil"
+	"github.com/ViaQ/logerr/log"
+	"math"
+	"os"
+	"path/filepath"
+	"time"
+	"github.com/fsnotify/fsnotify"
+)
+
+type Event = fsnotify.Event
+type Op = fsnotify.Op
+
+var (
+debugOn bool = true
+)
+
+const (
+	Create Op = fsnotify.Create
+	Write     = fsnotify.Write
+	Remove    = fsnotify.Remove
+	Rename    = fsnotify.Rename
+	Chmod     = fsnotify.Chmod
+)
+
+// Watcher is like fsnotify.Watcher but also notifies on changes to symlink targets
+type Watcher struct {
+	watcher *fsnotify.Watcher
+}
+
+
+func NewWatcher() (*Watcher, error) {
+	w, err := fsnotify.NewWatcher()
+	return &Watcher{watcher: w}, err
+}
+
+// Event returns the next event.
+func (w *Watcher) Event() (e Event, err error) {
+	return w.EventTimeout(time.Duration(math.MaxInt64))
+}
+
+// EventTimeout returns the next event or os.ErrDeadlineExceeded if timeout is exceeded.
+func (w *Watcher) EventTimeout(timeout time.Duration) (e Event, err error) {
+	var ok bool
+	select {
```

Comment:
I'm confused regarding this entire block.  It seems we should run through all these things in a prescribed fashion and early exist or other as needed if there is a timeout or error